### PR TITLE
feat: add dropdownWidth prop to Menu

### PR DESCRIPTION
**Pull request by mbylstra**
## A new `dropdownWidth` prop for the `Menu` component
`dropdownWidth="default"` (the default: fixed width at 248px, this is a non-breaking change)
https://dev.cultureamp.design/dst/menu-dropdownwidth-prop/storybook-static/?path=/story/menu-react--default-kebab
image

`dropdownWidth="contain"` (optional)
https://dev.cultureamp.design/dst/menu-dropdownwidth-prop/storybook-static/?path=/story/menu-react--dropdown-width-contain
image

**I also added some JSDocs to the Menu props while I was at it**

### CODE REVIEW

File: draft-packages/menu/KaizenDraft/Menu/Menu.tsx [hunk 1]
```diff
@@ -5,9 +5,33 @@ import MenuDropdown from "./MenuDropdown"
 const styles = require("./styles.scss")
 
 export type GenericMenuProps = {
```
Comment (mbylstra) [inline draft-packages/menu/KaizenDraft/Menu/Menu.tsx:7]: I also add some JSDoc's while I was at it

**Comment by mbylstra** [comment]
I'm totally open to changing the naming if you have better ideas!

File: draft-packages/menu/KaizenDraft/Menu/Menu.tsx [hunk 2]
```diff
@@ -5,9 +5,33 @@ import MenuDropdown from "./MenuDropdown"
...
+   * prop will not have any affect, as the state is handled internally to the component.
+   * @default: false
+   */
   menuVisible?: boolean
   automationId?: string
+
+  /**
+   * The content to appear inside the dropdown when it is open
+   */
   children: React.ReactNode
```
Comment (ActuallyACat) [inline draft-packages/menu/KaizenDraft/Menu/Menu.tsx:35]: I'm SO STOKED to see these comments

File: draft-packages/menu/KaizenDraft/Menu/Menu.tsx [hunk 3]
```diff
@@ -5,9 +5,33 @@ import MenuDropdown from "./MenuDropdown"
...
+   * Whether the menu is to be used on the left or right
+   * side of the viewport. If left, the left of the dropdown
+   * is aligned to the left of the button (and vice versa)
+   * @default "left"
+   */
   align?: "left" | "right"
+
+  /**
+   * The width of the dropdown.
+   * "default": a fixed width of 248px
```
Comment (ActuallyACat) [inline draft-packages/menu/KaizenDraft/Menu/Menu.tsx:18]: Why 248px? Am I right in thinking that it was already doing this?

Comment (mbylstra) [reply draft-packages/menu/KaizenDraft/Menu/Menu.tsx:18]: Yeah, that was already there. Not sure why it was such as specific width, but I don't want to break anything existing.

Comment (ActuallyACat) [reply draft-packages/menu/KaizenDraft/Menu/Menu.tsx:18]: Yup, that makes sense.

**Pull request merged**